### ISSUE TITLE
Create group chats based on proximity

### ISSUE THREAD

**Issue by dvilchez** [reTHINK-project/dev-participate#16]
I was struggling with this requirement for a while and I can't figure out how to accomplish it.

Location Hyperty is responsible of taking the position and share it with any hyperty subscribed to the dataobject but how can I get all Location Hyperties on the system, filter them by proximity and get their identity?

As far as I know the communication channels are:

developer->hyperty, using hyperty instance.
hyperty<->hyperty, through syncher
hyperty<->core components/proto stubs, through messaging 

If the position info is not in any other place than the hyperty Do I need create a dataobject with each one location hyperty on the system? I don't think so.

Is this a responsibility of another component on the system?

What am I doing wrong?

@pchainho @shumy @vitormsilva

**Comment by antonroman** [comment]
After the discussion today, we conclude that the best realistic way to create groups based on proximity would be to add a location attribute to the Communication Data Schema, when the organizer creates the group it adds its location. When the rest of the people receives the invitation to join they can check the distance to their location and accept if they consider they should join.
@dvilchez what's your view on this approach?

**Comment by dvilchez** [comment]
Yes, it is very close to the approach I implemented but the problem here is the user wants to invite people in an area and it is a previous step.

**Comment by pchainho** [comment]
Instead of using a new attribute to Communication data object we could also use the ["Contextual Communication"](https://github.com/reTHINK-project/dev-smart-contextual-assistance-app/tree/master/docs) object used by our App.
